fix: added dafault image for avatar to avoid error

YuraPetrovskyi opened the pull request:
# Description

Relates #394

This Pull Request fixes a runtime error that occurred when the partnerAvatar value was null in the ConversationCard component. The fix introduces a default profile image (/default-profile.png) to ensure that the application gracefully handles cases where a user has not set a profile picture.

No new dependencies were introduced.

### Files changed

- ConversationCard.tsx 
Updated the useState initialization for imgSrc to assign a default image (/default-profile.png) when partnerAvatar is null.

### UI changes

No visible UI changes as the fix ensures a fallback image is displayed when the profile picture is not set.

### Changes to Documentation

No updates to documentation were necessary for this change.

# Tests

The changes were tested locally, and all existing tests passed. Since this is a minor fix, no additional tests were added.

YuraPetrovskyi committed:
fix: added dafault image for avatar to avoid error

relates #394

camelPhonso commented on components/messaging/ConversationCard.tsx:
```diff
@@ -39,7 +39,7 @@ const ConversationCard: React.FC<ConversationCardProps> = ({
 }) => {
   const isBreakpoint = useMediaQuery(1000);
-  const [imgSrc, setImgSrc] = useState(partnerAvatar);
+  const [imgSrc, setImgSrc] = useState(partnerAvatar || '/default-profile.png');
```
watch out for this one! 

The `OR` operator is not what we'd want for this, it should be a `nullish coalescent operator` (`??`) to return the string '/default-profile.png' if `partnerAvatar` is`null`

_however_

what's really happening here is that on line 62 the `Image` component has an `onError` attribute that is intercepting the error and setting the `src` to the default image.

Right now the error shows up on the console because it's that error that's triggering the default avatar being displayed, but if you comment out the `onError` attribute you will see that we don't get any image anymore.

Unfortunately I don't think this issue is a frontend one, we should think of a more robust solution for our database really - ensuring images for our `dev` are persistent across separate environments but also injecting the default profile picture into the users' data if we see they don't have one. That way we would find the error once and clear it permanently and remove the weight from our frontend.

Until then, I think the component is handling it gracefully enough

YuraPetrovskyi replied:
Hi @camelPhonso ,
Thank you so much for reviewing my code and for your detailed explanation. I really appreciate the insights you provided and the time you took to highlight the potential improvements.

> what's really happening here is that on line 62 the `Image` component has an `onError` attribute that is intercepting the error and setting the `src` to the default image.
> 
> Right now the error shows up on the console because it's that error that's triggering the default avatar being displayed, but if you comment out the `onError` attribute you will see that we don't get any image anymore.
> 
Regarding your point about the error only showing up in the console and being handled by the onError attribute, I think there’s an issue that happens earlier. The application actually **crashes before** the `onError` function is triggered because we're trying to set `src={null}`, which is invalid for the src attribute of the <Image> component. Could you confirm if I’m understanding this correctly?
> . That way we would find the error once and clear it permanently and remove the weight from our frontend.
> 
I also wanted to ask your opinion on a potential solution you mentioned regarding handling this in the database. Would it be a good idea to set a DEFAULT value for the avatar column, like this:
```
ALTER TABLE profiles  
ALTER COLUMN avatar SET DEFAULT '/default-profile.png';
```

camelPhonso replied:
> I also wanted to ask your opinion on a potential solution you mentioned regarding handling this in the database. Would it be a good idea to set a DEFAULT value for the avatar column, like this:

Hey @YuraPetrovskyi - I think adding a default value makes perfect sense.

The example you gave would need to be run on a migration to fix any data already present and then we'd need to refactor the user signup process to insert this value if there is no image uploaded. It'd be a good shout to then also make sense that column is not nullable in the database and that all our types recording these entries don't have the image source as nullable either.

This would probably stop any issues related with loading the images, but if we still see an error popup we'll know to look for a different cause. In any case, I think this change would always be a gain for us so it'd be worth implementing either way 👌

camelPhonso reviewed (changes requested):
We should try to fix this one on the backend probably 🤔

YuraPetrovskyi committed:
fix: replaced OR operator with nullish coalescing operator for avatar initialization

relates #394

YuraPetrovskyi committed:
fix: updated seed data to include default avatar

relates #394

YuraPetrovskyi committed:
feat: added default value for avatar column in profiles table

relates #394

YuraPetrovskyi commented:
Hi @camelPhonso ,
Thank you for your detailed feedback and suggestions. I've implemented the following changes based on your recommendations:

**Updated the imgSrc initialization**:
Replaced the OR operator (||) with the nullish coalescing operator (??) in ConversationCard.tsx as a precaution to safeguard against errors, even though the subsequent steps should prevent such issues more reliably.

**Updated seed data:**
In the seed.sql file, I replaced NULL values in the avatar column with '/default-profile.png' for users without an avatar. This ensures consistency with the new logic.

**Set a default value in the database:**
Added a migration to set '/default-profile.png' as the default value for the avatar column in the profiles table. This guarantees all new records without a specified avatar will automatically use the default profile image.

**Updated existing values in the main Supabase database:**
To maintain consistency, I ran the following SQL query on the main Supabase database to update all existing NULL values in the avatar column:
```
UPDATE "public"."profiles"
SET "avatar" = '/default-profile.png'
WHERE "avatar" IS NULL;
```
Let me know if there's anything else you'd like me to adjust or consider!

ShaughnAnderson94 reviewed (approved):
Hey I think you got it working and your adjustments based on Alphonso's feedback seem like they will address all of his concerns moving stuff to being handled on the backend. 🍾  (Although I am struggling to resolve the conversation so @camelPhonso might need to do that)

nichgalzin reviewed (approved):
Great work

camelPhonso reviewed (approved):
Amazing! Thank you for the detailed update Yurii.

Sorry for the delay reviewing - this is great stuff :)
